lib/ASTGen/Sources/ASTGen/ASTGen.swift

```diff
@@ -150,106 +133,215 @@ extension ASTGenVisitor {
+        return generate(node)
+    case .dictionaryType(let node):
+        return generate(node)
+    case .metatypeType(let node):
+        return generate(node)
+    case .implicitlyUnwrappedOptionalType(let node):
+        return generate(node)
+    case .packExpansionType(let node):
+        return generate(node)
+    case .optionalType(let node):
+        return generate(node)
+    case .tupleType(let node):
+        return generate(node)
+    case .compositionType(let node):
+        return generate(node)
+    case .functionType(let node):
+        return generate(node)
+    case .namedOpaqueReturnType(let node):
+        return generate(node)
+    case .someOrAnyType(let node):
+        return generate(node)
+    case .attributedType(let node):
+        return generate(node)
+    case .codeBlock(let node):
+        return generate(node)
+    case .expressionStmt(let node):
+        return generate(node)
+    case .returnStmt(let node):
+        return generate(node)
+    case .functionParameter(let node):
+        return generate(node)
+    case .enumCaseParameter(let node):
+        return generate(node)
+    case .functionParameterClause(let node):
+        return generate(node)
+    case .enumCaseParameterClause(let node):
+        return generate(node)
+    case .stringLiteralExpr(let node):
+        return generate(node)
+    case .integerLiteralExpr(let node):
+        return generate(node)
+    case .booleanLiteralExpr(let node):
+        return generate(node)
+    case .arrayExpr(let node):
+        return generate(node)
+    case .nilLiteralExpr(let node):
+        return generate(node)
+    case .genericParameterClause(let node):
+        return generate(node)
+    case .genericParameter(let node):
+        return generate(node)
+    case .genericWhereClause(let node):
+        return generate(node)
+    default:
+        fatalError("not implemented")
+    }
   }
 }
 
 // Forwarding overloads that take optional syntax nodes. These are defined on demand to achieve a consistent
 // 'self.visit(<expr>)' recursion pattern between optional and non-optional inputs.
 extension ASTGenVisitor {
   @inline(__always)
-  func visit(_ node: TypeSyntax?) -> ASTNode? {
+  func generate(_ node: TypeSyntax?) -> ASTNode? {
     guard let node else {
       return nil
     }
 
-    return self.visit(node)
+    return self.generate(node)
   }
 
   @inline(__always)
-  func visit(_ node: ExprSyntax?) -> ASTNode? {
+  func generate(_ node: ExprSyntax?) -> ASTNode? {
     guard let node else {
       return nil
     }
 
-    return self.visit(node)
+    return self.generate(node)
   }
 
   @inline(__always)
-  func visit(_ node: (some SyntaxChildChoices)?) -> ASTNode? {
+  func generate(_ node: (some SyntaxChildChoices)?) -> ASTNode? {
     guard let node else {
       return nil
     }
 
     // This call recurses without disambiguation.
-    return (self.visit as (_) -> ASTNode)(node)
+      return self.generate(node)
```

Comment:
AFAICT this causes an infinite recursion. If you add `func generate(_ node: some SyntaxChildChoices) -> ASTNode` as suggested above, then you should be able to do 

```swift
return self.generate(node) as ASTNode
```